code/_helpers/russian.dm

```diff
@@ -67,24 +72,24 @@
 
 //input
 
-/proc/input_cp1251(var/mob/user = usr, var/message, var/title, var/default, var/type = "message")
+/proc/input_cp1251(var/mob/user = usr, var/message, var/title, var/default, var/type = "message", var/prepare_to_browser = FALSE)
 	var/msg = ""
 	switch(type)
 		if("message")
-			msg = input(user, message, title, edit_cp1251(default)) as null|message
+			msg = input(user, message, title, edit_cp1251(default)) as message
 		if("text")
-			msg = input(user, message, title, default) as null|text
-	msg = russian_to_cp1251(msg)
+			msg = input(user, message, title, default) as text
+	msg = russian_to_cp1251(msg, prepare_to_browser)
 	return post_edit_cp1251(msg)
 
-/proc/input_utf8(var/mob/user = usr, var/message, var/title, var/default, var/type = "message")
+/proc/input_utf8(var/mob/user = usr, var/message, var/title, var/default, var/type = "message", var/prepare_to_browser = FALSE)
 	var/msg = ""
 	switch(type)
 		if("message")
-			msg = input(user, message, title, edit_utf8(default)) as null|message
+			msg = input(user, message, title, edit_utf8(default)) as message
 		if("text")
-			msg = input(user, message, title, default) as null|text
-	msg = russian_to_utf8(msg)
+			msg = input(user, message, title, default) as text
+	msg = russian_to_utf8(msg, prepare_to_browser)
```

Comment:
Why ` as null|text` was replaced with ` as text` (same for `message`)?

Reply:
They were lost during port, but i see no reason to consider it as null

Reply:
`as null` add `cancel` button to input dialogue